### ISSUE TITLE
[Bug]: Nested UiMenus cannot be navigated when inside a focus trap

### ISSUE THREAD

**Issue by spaceemotion** [radix-vue/radix-vue#816]
### Environment

```bash
Development/Production OS: WSL2 Ubuntu 22.2
Node version: 20
Package manager: pnpm@8.15
Radix Vue version: 1.6.1
Vue version: 3.421
CSS framework: tailwindcss@3.4.3
Client OS: Windows 10
Browser: Edge 122, Safari 17.4
```

### Link to minimal reproduction

(following soon, still preparing)

### Steps to reproduce

1. Put a UiMenu hierarchy inside a `UseFocusTrap` component from vueuse
2. Click the trigger button inside the focus trap to open the menu
3. When no trying to open sub menus they disappear once the first menu layer gets exited. Keyboard navigation is also broken.

### Describe the bug

For some reason the update to 1.6.0 (1.6.1) broke something in the handling of nested UI menus. From what I can tell it has something to do with focus states, but checking the commit logs I can't figure out where that error got introduced. 

In my case I got a FocusTrap from vueuse (via the `UseFocusTrap` component). Since the UiMenu is outside of that hierarchy I guess it messes up the focus detection somehwere?

In 1.5.3 this all worked fine.

### Expected behavior

The submenus just open, and keyboard navigation also works.

### Context & Screenshots (if applicable)

https://github.com/radix-vue/radix-vue/assets/429147/368cab02-d3d3-439c-8f88-b5d54ea765a9

**Comment by zernonia** [comment]
@spaceemotion  could you create a minimal repro for this? It will be hard to trace down the issue without that.
Cause in docs submenu still works nicely  https://www.radix-vue.com/components/dropdown-menu.html

**Comment by spaceemotion** [comment]
@zernonia yeah that was the plan - had to stop for the day before i could finish it. need to see how to replicate this in a small and clear manner :)

I had to downgrade back to 1.5.3 because of this, but i noticed that even in that version neither the hover states nor keyboard navigation works. The menu opens properly, but it can't be accessed. Best guess is that 1.6.0 actually fixes an issue, that I kind of relied on previously? Currently what I figure is that since the menu hierarchy is outside of the focus trap's DOM node there is that weird behavior because of it.

**Comment by zernonia** [comment]
Ok no worry @spaceemotion . I had a hunch which changes causes this. Let's see on version `1.6.2` (releasing real soon) 😁

**Comment by zernonia** [comment]
@spaceemotion  thanks for the support. Can you try `1.6.2` and see if this bug still valid? 😁

**Comment by spaceemotion** [comment]
@zernonia no problem! I just updated the app to use 1.6.2 and it seems to be back on the old behavior. so that's good. i guess the hover and keyboard events still don't trigger because of the focus trap, but mouse clicks on the menu items are registered properly again.

Do you still want me to prepare a sandbox to replicate this?

**Comment by zernonia** [comment]
Yeah please do @spaceemotion ! We wanna see what's the issue you are facing, and possible use case. Maybe update the title to better describe the issue you are having 😁

**Comment by zernonia** [comment]
Closed as stale. Do reopen the ticket with reproduction ya 😁
